### PR TITLE
[LoongArch64] Modify dwarfReg in native llvm-libunwind to be consistent with the LoongArch manual.

### PR DESCRIPTION
**Pull request by sunlijun-610**
Change the `UNW_LOONGARCH` of `F0-F31` from 32-63 to 64-95.

### CODE REVIEW

**Review by shushanhf — APPROVED** [review]
LGTM
Thanks

File: src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfExpressionBuilder.cs [hunk 1]
```diff
@@ -162,7 +162,10 @@ public static int DwarfRegNum(TargetArchitecture architecture, int regNum)
                     };
 
                 case TargetArchitecture.LoongArch64:
+                    Debug.Assert(regNum <= 64);
```
Comment (am11) [inline src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfExpressionBuilder.cs:165]: `using System.Diagnostics;` at the top of this file is needed.

Comment (sunlijun-610) [reply src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfExpressionBuilder.cs:165]: Oh, I forgot it. So should I delete this line or add `using System.Diagnostics` at the top?
Thanks for your help!

Comment (am11) [reply src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfExpressionBuilder.cs:165]: Add after https://github.com/dotnet/runtime/blob/926885adc308007934a97e6d36b59cfb7a535a66/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfExpressionBuilder.cs#L5 (so they are alphabetized)

Comment (sunlijun-610) [reply src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfExpressionBuilder.cs:166]: OK, thanks a lot!

File: src/native/external/llvm-libunwind/include/libunwind.h [hunk 2]
```diff
@@ -1286,38 +1286,38 @@ enum {
...
-  UNW_LOONGARCH_F23 = 55,
-  UNW_LOONGARCH_F24 = 56,
-  UNW_LOONGARCH_F25 = 57,
-  UNW_LOONGARCH_F26 = 58,
-  UNW_LOONGARCH_F27 = 59,
-  UNW_LOONGARCH_F28 = 60,
-  UNW_LOONGARCH_F29 = 61,
-  UNW_LOONGARCH_F30 = 62,
-  UNW_LOONGARCH_F31 = 63,
+  UNW_LOONGARCH_F0 = 64,
```
Comment (jkotas) [inline src/native/external/llvm-libunwind/include/libunwind.h:1289]: This is vendored project from https://github.com/llvm/llvm-project/blob/main/libunwind/include/libunwind.h#L1269 . Is this a bug in the upstream project? If yes, could you please make sure that this is upstreamed to https://github.com/llvm/llvm-project as well?

Comment (shushanhf) [reply src/native/external/llvm-libunwind/include/libunwind.h:1289]: Thanks  very much~!
You are right.
The `jit/unwindloongarch64.cpp` is too old.

we have to follow the latest la-abi speci https://github.com/loongson/la-abi-specs/blob/release/ladwarf.adoc
And we have to modify this file https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/unwindarm64.cpp#L21

Also thanks @filipnavara 

We will close this PR and push a new PR to modify the unwindarm64.cpp.

**Comment by sunlijun-610** [comment]
The new PR for modifying jit/unwindloongarch64.cpp has been submitted to https://github.com/dotnet/runtime/pull/106382

**Comment by sunlijun-610** [comment]
The new PR for modifying jit/unwindloongarch64.cpp has been submitted to https://github.com/dotnet/runtime/pull/106382
And this PR can be closed.

**Pull request closed without merging**